CCScriptWriter/CCScriptWriter.py

```diff
@@ -478,16 +379,60 @@ def loadDialogue(self, loadCoilSnake=False):
                                    FormatHex(self.data[a + 7]),
                                    FormatHex(self.data[a + 8])))
                 t = 1
+            else:
+                assert False
             m = self.dataFiles[address]
             h = hex(address)
             self.asmPointers[a] = ["{}.l_{}".format(m, h), t]
 
+    def loadCoilSnakeDialogue(self):
+        "Load pointers from the CoilSnake project."
+        o = os.path.join(self.outputDirectory, os.path.pardir)
+        project = os.path.join(o, "Project.snake")
+        try:
+            with open(project, "r", encoding="utf8"):
+                pass
+        except IOError:
+            print("Failed to open \"{}\". Invalid CoilSnake project. "
+                    "Aborting.".format(project))
+            sys.exit(1)
```

Comment:
```suggestion
            raise RuntimeError("Failed to open \"{}\". Invalid CoilSnake project. "
                    "Aborting.".format(project))
```
sys.exit [raises a `SystemExit` exception](https://docs.python.org/3/library/sys.html#sys.exit), which it is technically possible to catch, but most code won't catch it [even if it tries to catch `Exception`](https://docs.python.org/3/library/exceptions.html#SystemExit). (And if you look at CoilSnake's code, [it does exactly that](https://github.com/pk-hack/CoilSnake/blob/d806fa6e094000156184af8cbf8fb973d3dc34a8/coilsnake/ui/common.py#L245-L251). This seems to be the root cause of the issue that pk-hack/CoilSnake#289 was trying to solve.) Exiting from the whole interpreter isn't something that library code should be trying to do.

Reply:
I changed these to ValueErrors instead, thanks for pointing these out.